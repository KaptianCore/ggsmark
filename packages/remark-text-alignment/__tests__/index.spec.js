import remark from 'remark'
import html from 'remark-html'
import alignment from '..'
import dedent from 'dedent'

describe('align text with inline style', () => {
  test('should align single line', () => {
    // Arrange
    let string = dedent`
    <-LeftText<-
    `

    // Act
    let result = remark()
      .use(html)
      .use(alignment, { useInlineStyles: true })
      .processSync(string)
      .toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should align multi line', () => {
    // Arrange
    let string = dedent`
    <-This is a multiple line texts
    2nd line here<-
    `

    // Act
    let result = remark()
      .use(html)
      .use(alignment, { useInlineStyles: true })
      .processSync(string)
      .toString()

    // Assert
    expect(result).toMatchSnapshot()
  })
})

describe('align text with class names', () => {
  test('should align single line', () => {
    // Arrange
    let string = dedent`
    <-LeftText<-
    `

    // Act
    let result = remark()
      .use(html)
      .use(alignment)
      .processSync(string)
      .toString()

    // Assert
    expect(result).toMatchSnapshot()
  })

  test('should align multi line', () => {
    // Arrange
    let string = dedent`
    <-This is a multiple line texts
    2nd line here<-
    `

    // Act
    let result = remark()
      .use(html)
      .use(alignment)
      .processSync(string)
      .toString()

    // Assert
    expect(result).toMatchSnapshot()
  })
})
