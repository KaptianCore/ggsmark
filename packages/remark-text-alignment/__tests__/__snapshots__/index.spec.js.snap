// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align text with class names should align multi line 1`] = `
"<div class=\\"align-left\\"><p>This is a multiple line texts
2nd line here</p></div>
"
`;

exports[`align text with class names should align single line 1`] = `
"<div class=\\"align-left\\"><p>LeftText</p></div>
"
`;

exports[`align text with inline style should align multi line 1`] = `
"<div style=\\"text-align: left\\"><p>This is a multiple line texts
2nd line here</p></div>
"
`;

exports[`align text with inline style should align single line 1`] = `
"<div style=\\"text-align: left\\"><p>LeftText</p></div>
"
`;
